src/components/common/PopoverToggle/index.js

```diff
@@ -26,76 +26,66 @@ Popover.defaultProps = {
   active: false,
 };
 
-class PopoverToggle extends Component {
-  static propTypes = {
-    children: PropTypes.node.isRequired,
-    className: PropTypes.string,
-    history: PropTypes.object.isRequired,
-    popoverClassName: PropTypes.string,
-    popoverContent: PropTypes.node,
-  };
+const PopoverToggle = ({
+  className,
+  popoverClassName,
+  popoverContent,
+  children,
+}) => {
+  const history = useHistory();
+  const dropdown = useRef();
 
-  static defaultProps = {
-    className: '',
-    popoverClassName: '',
-    popoverContent: '',
-  };
+  const [isOpen, setIsOpen] = useState(false);
+  const close = useCallback(() => setIsOpen(false), []);
 
-  constructor(props) {
-    super(props);
-    this.toggle = this.toggle.bind(this);
-    this.close = this.close.bind(this);
-    this.outsideHook = this.outsideHook.bind(this);
-    this.unlisten = () => {};
-  }
+  // close when click outside of dropdown
+  const outsideHandler = useCallback(
+    e => {
+      if (!dropdown.current.contains(e.target)) {
+        if (isOpen) {
+          close();
+        }
+      }
+    },
+    [close, isOpen],
+  );
+  useEffect(() => {
+    document.addEventListener('click', outsideHandler);
+    return () => {
+      document.removeEventListener('click', outsideHandler);
+    };
+  }, [outsideHandler]);
 
-  state = {
-    isOpen: false,
-  };
+  // close when routing change
+  useEffect(() => {
+    return history.listen(close);
```

Comment:
Suggestion: `return` 只用在 deconstruct handler

```suggestion
    history.listen(close);
```

Reply:
see https://github.com/remix-run/history/blob/v4/docs/GettingStarted.md#basic-usage

`history.listen` will return an unlisten handler